Actions/PullPowerPlatformChanges/action.yaml

```
@@ -0,0 +1,142 @@
+        }
+        catch {
+          Write-Host "::ERROR::Unexpected error when running action. Error Message: $($_.Exception.Message.Replace("`r",'').Replace("`n",' ')), StackTrace: $($_.ScriptStackTrace.Replace("`r",'').Replace("`n",' <- '))";
+          exit 1
+        }
+
+    - name: Check out AL-Go Actions
+      uses: actions/checkout@v4
+      with:
+        repository: ${{ env.actionsRepo }}
+        ref: ${{ env.actionsRef }}
+        path: ${{ env.actionsPath }}
+
+    - name: Parse DeployToSettings and AuthContext
+      id: ReadPowerPlatformSettings
+      uses: ./_AL-Go/Actions/ReadPowerPlatformSettings
+      with:
+        shell: ${{ inputs.shell }}
+        deploymentEnvironmentsJson: ${{ inputs.deploymentEnvironmentsJson }} 
+        environmentName: ${{ inputs.environmentName }}
+
+    - name: Set up new branch for changes
+      shell: ${{ inputs.shell }}
+      env:
+        _actor: ${{ inputs.actor }}
+        _token: ${{ inputs.token }}
+        _updateBranch: ${{ inputs.updateBranch }}
+        _directCommit: ${{ inputs.directCommit }}
+      run: |
+        $errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
+        try {
+          ${{ github.action_path }}/GitCloneReponsitory.ps1 -actor $ENV:_actor -token $ENV:_token -updateBranch $ENV:_updateBranch -directCommit ($ENV:_directCommit -eq 'true')
+        }
+        catch {
+          Write-Host "::ERROR::Unexpected error when running action. Error Message: $($_.Exception.Message.Replace("`r",'').Replace("`n",' ')), StackTrace: $($_.ScriptStackTrace.Replace("`r",'').Replace("`n",' <- '))";
+          exit 1
+        }
+
+    - name: Export Solution (username)
+      if: steps.ReadPowerPlatformSettings.outputs.ppUserName != ''
+      uses: microsoft/powerplatform-actions/export-solution@v1
+      with:
+        user-name: ${{ steps.ReadPowerPlatformSettings.outputs.ppUserName }}
+        password-secret: ${{ steps.ReadPowerPlatformSettings.outputs.ppPassword }}
+        environment-url: ${{ steps.ReadPowerPlatformSettings.outputs.ppEnvironmentUrl }}
+        solution-name: ${{ inputs.solutionFolder }}
+        solution-output-file: ${{ env.clonedRepoPath }}/${{ inputs.solutionFolder }}.zip
+
+    - name: Export Solution (application ID)
+      if: steps.ReadPowerPlatformSettings.outputs.ppApplicationId != ''
+      uses: microsoft/powerplatform-actions/export-solution@v1
+      with:
+        tenant-id: ${{ steps.ReadPowerPlatformSettings.outputs.ppTenantId }}
+        app-id: ${{ steps.ReadPowerPlatformSettings.outputs.ppApplicationId }}
+        client-secret: ${{ steps.ReadPowerPlatformSettings.outputs.ppClientSecret }}
+        environment-url: ${{ steps.ReadPowerPlatformSettings.outputs.ppEnvironmentUrl }}
+        solution-name: ${{ inputs.solutionFolder }}
+        solution-output-file: ${{ env.clonedRepoPath }}/${{ inputs.solutionFolder }}.zip
+
+    - name: unpack-solution action
```

Name could be better. E.g.
```suggestion
    - name: Unpack Solution
```